intro/sdk_code_guidelines.md

```diff
@@ -0,0 +1,159 @@
+The copyright statement must appear in every source file with the `$today.year` [Velocity](https://www.jetbrains.com/help/idea/copyright-profiles.html) template resolved.
+
+## Plugin ID Conventions
+The plugin ID (`<id>` in `plugin.xml`) always begins with `org.intellij.sdk`.
+The plugin ID is known as the `Group ID` in the Gradle plugin wizard.
+
+For _basic_ code samples, it is not necessary to include "basic" in the plugin ID.
+A plugin like `facet_basics` has the ID `org.intellij.sdk.facet`.
+
+## Plugin Package Names
+Packages in plugins should begin with the plugin ID.
+If there is only one package in a plugin, then the package name is the same as the plugin ID.
+
+## Plugin Directory Structure
+SDK sample code should have a standard directory footprint.
+Standardized structure not only makes the samples easier to navigate and understand, but it builds on the default Gradle plugin project structure.
+The following is the directory structure for a `basic_foo` plugin.
+```text
+code_samples/
+  basic_foo/
+    gradle/
+    src/
+      main/
+        java/
+          icons/
+            BasicFooIcons.java
+        resources/
+          icons/
+            sdk.svg           # The standard SDK icon for menus, etc.
+          META-INF/
+            plugin.xml        # The plugin configuration file
+            pluginIcon.svg    # The standard SDK plugin icon
+      test/                   # Omit if there are no tests
+        java/
+        resources/
+    build.gradle
+    gradlew
+    gradle.bat
+    settings.gradle
+```
+
+## build.gradle Conventions 
+New SDK code samples should be developed [using Gradle](/tutorials/build_system.md). 
+As of this writing, the use of Gradle in SDK code samples still relies heavily on the `plugin.xml` for specifying the plugin configuration.
+At a later, second phase, the SDK code samples will transition to rely more on the Gradle configuration. 
+
+The default contents of a `build.gradle` file are produced by the [Gradle project wizard](/tutorials/build_system/prerequisites.md#create-a-plugin-project-from-scratch). 
+A consistent structure for an SDK code sample's `build.gradle` file is important for clarity and is based on the default produced by the project wizard. 
+Comments in SDK code sample `build.gradle` files should only draw attention to the parts of the Gradle configuration that are unique for a plugin.
+
+For SDK code samples a few alterations are needed to the default build.gradle file produced by the plugin wizard:
+* Add the following statements to the [Setup DSL](https://github.com/JetBrains/gradle-intellij-plugin/blob/master/README.md#setup-dsl) (`intellij{}`) section:
+  ```groovy
+      // Prevents patching <idea-version> attributes in plugin.xml
+      updateSinceUntilBuild = false 
+      // Define a shared sandbox directory for running code sample plugins within an IDE.
+      sandboxDirectory = file("${project.projectDir}/../_Build/idea-sandbox")
```

Comment:
where does `_Build` come from?

Reply:
"_Build" = directory that holds the sandbox directory. Appears at the top of the code_samples directory for convenience. code_samples/action_basics/build is the default gradle output directory where everything else is located.